Dreier Neuerscheinungsdienst.js

```diff
@@ -0,0 +1,115 @@
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+
+function detectWeb(doc, url) {
+	if (url.includes('/detail/ISBN')) {
+		return "book";
+	}
+	else if (getSearchResults(doc, true)) {
+		return "multiple";
+	}
+	return false;
+}
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	// TODO: adjust the CSS selector
+	var rows = doc.querySelectorAll('h4 a');
+	for (let row of rows) {
+		let href = row.href;
+		let title = ZU.trimInternal(row.textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (items) ZU.processDocuments(Object.keys(items), scrape);
+		});
+	}
+	else {
+		scrape(doc, url);
+	}
+}
+
+function scrape(doc) {
+	var item = new Zotero.Item("book");
+	// Set the title
+	item.title = ZU.xpathText(doc, '//h1[@class="biblioTitle"]');
+	if (ZU.xpathText(doc, '//div[@class="titles"]/div[@class="biblioSubTitle"]')) item.title += ": " + ZU.xpathText(doc, '//div[@class="titles"]/div[@class="biblioSubTitle"]');
+	// ISBN
+	item.ISBN = (doc.querySelector('.biblioId .value').textContent) ? doc.querySelector('.biblioId .value').textContent : "";
+	// Publisher
+	item.publisher = (doc.querySelector('.biblioPublisher .value').textContent) ? doc.querySelector('.biblioPublisher .value').textContent : "";
+	// Publisher Location
+	item.place = (doc.querySelector('.biblioPublicationTown .value').textContent) ? doc.querySelector('.biblioPublicationTown .value').textContent : "";
+	// Publication Date
+	item.date = (doc.querySelector('.biblioPublishingYear .value').textContent) ? doc.querySelector('.biblioPublishingYear .value').textContent : "";
+	// Number of pages
+	item.numPages = (doc.querySelector('.biblioPages .value').textContent) ? doc.querySelector('.biblioPages .value').textContent : "";
+	// Abstract
+	item.abstractNote = (doc.querySelector('.description .blurb .value').textContent) ? doc.querySelector('.description .blurb .value').textContent : "";
+	// Get Creators
+	var creators = doc.querySelectorAll('.authorMain .biblioAuthor');
+	for (let creator of creators) {
+		let creatorName = creator.querySelector('.value').textContent;
+		let creatorRole;
+		
+		// currently, we only check for editors; everything else will be treated as authors
+		if (creator.textContent.includes("Hrsg.")) {
+			Z.debug(creatorName + " ist Herausgeber");
```

Comment:
I know this translator targets a German publisher and will probably mostly have German users, but in the spirit of the code and comments being in English, I think the debug messages should be, too.

Reply:
I've removed the debug messages entirely. I've also made the tests more robust by adding French and English terms. Now creators work now also in the English and French version of the webpage.